packages/clay-form/src/SelectBox.tsx

```diff
@@ -212,7 +214,13 @@ const ClaySelectBox: React.FunctionComponent<IProps> = ({
 							<ClayButtonWithIcon
 								aria-label={ariaLabels.reorderUp}
 								className="reorder-button reorder-button-up"
-								disabled={!value.length}
+								disabled={
+									!value.length ||
+									selectedIndex === 0 ||
+									selectedIndexes.includes(0) ||
+									!items.length ||
+									selectedIndexes.length === 0
+								}
```

Comment:
This expression is hard to follow, I think it'd be best to put into some named variables in order to clean it up. I also noticed that `selectedIndexes` is directly related to `value`, so you shouldn't need to have checks for both.

```jsx
const noItems = !items.length;
const noItemsSelected = !selectedIndexes.length;
const firstItemSelected = selectedIndex === 0 || selectedIndexes.includes(0);
```

Reply:
You're right, I got caught up in making it work, so I didn't think much about readability.